Fix mistakes with HPC and Manual instructions

milljm opened the pull request:
Fixing mistakes with the HPC and Manual instruction pages.
Remove bread crumb pointing to an outdated index.

Closes #12525

Commit:
Fix mistakes with HPC and Manual instructions

Closes #12525

Commit:
Adjust location of Syntax/Source documentation

An email suggested these two items should be at the top.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/260042/) on 4a78235 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/12526/site)

This comment will be updated on new commits.

Commit:
LLVM/MPICH is hard

Alert the user they are treading in difficult waters when choosing to build the LLVM/MPICH stack. Because it's hard. Really. I barely understand it myself.
Add a sort of page break between adjacent paragraph siblings within our alerts.
Add instructions on skipping MPICH for GCC if its not needed.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/260063/) on 19ad2da : invalidated by @aeslaughter

Rebuild of site
